backend/ml/inputs.py

```diff
@@ -41,63 +46,75 @@ def ratings_properties(self) -> pd.DataFrame:
             * `user_id`: int
             * `entity_id`: int or str
             * `is_public`: bool
-            * `is_trusted`: bool
-            * `is_supertrusted`: bool
+            * `is_scaling_calibration_user`: bool
+            * `trust_score`: float
         """
         raise NotImplementedError
 
 
 class MlInputFromPublicDataset(MlInput):
-    def __init__(self, csv_file):
-        self.public_dataset = pd.read_csv(csv_file)
-        self.public_dataset.rename(
-            {"video_a": "entity_a", "video_b": "entity_b"}, axis=1, inplace=True
-        )
-        self.public_dataset["user_id"], self.user_indices = self.public_dataset[
-            "public_username"
-        ].factorize()
+    def __init__(self, dataset_zip: Union[str, BinaryIO]):
+        if isinstance(dataset_zip, str) and (
+            dataset_zip.startswith("http://") or dataset_zip.startswith("https://")
+        ):
+            dataset_zip, _headers = urlretrieve(dataset_zip)  # nosec B310
+
+        with zipfile.ZipFile(dataset_zip) as zip_file:
+            zip_root_dir = next(path for path in zipfile.Path(zip_file).iterdir() if path.is_dir())
+
+            with (zip_root_dir / "comparisons.csv").open() as comparison_file:
+                self.comparisons = pd.read_csv(comparison_file)
+                self.comparisons.rename(
+                    {"video_a": "entity_a", "video_b": "entity_b"}, axis=1, inplace=True
+                )
 
-    def get_comparisons(
-        self, trusted_only=False, criteria=None, user_id=None
-    ) -> pd.DataFrame:
-        dtf = self.public_dataset.copy(deep=False)
+            with (zip_root_dir / "users.csv").open() as users_file:
+                self.users = pd.read_csv(users_file)
+                self.users.index.name = "user_id"
+
+            username_to_user_id = pd.Series(
+                data=self.users.index, index=self.users["public_username"]
+            )
+            self.comparisons = self.comparisons.join(username_to_user_id, on="public_username")
+
+    def get_comparisons(self, criteria=None, user_id=None) -> pd.DataFrame:
+        dtf = self.comparisons.copy(deep=False)
         if criteria is not None:
             dtf = dtf[dtf.criteria == criteria]
         if user_id is not None:
             dtf = dtf[dtf.user_id == user_id]
+        dtf["weight"] = 1
```

Comment:
I understand why users have a trust_score, and why user ratings have weights. But not why comparisons have weights. Should we keep this concept ?